Sources/Logging/Logging.swift

```diff
@@ -96,6 +96,40 @@ public struct Logger {
     }
 }
 
+extension Logger {
+    @inlinable
+    func log(level: Logging.Level, message: @autoclosure () -> String, error: Error, file: StaticString = #file, function: StaticString = #function, line: UInt = #line) {
+        if self.logLevel <= level {
+            self.handler.log(level: level, message: message(), error: error, file: file, function: function, line: line)
+        }
+    }
+
+    @inlinable
+    public func trace(_ message: @autoclosure () -> String, error: Error, file: StaticString = #file, function: StaticString = #function, line: UInt = #line) {
```

Comment:
why do we add all those methods again? They're already on `Logger`, we can just do `error: Error? = nil` no?

Reply:
we should never overload methods, that always leads to massive issues

Reply:
we can (see my original listed alternative). i was wondering which is the best approach. adding error: Error? = nil to the existing API is easier but makes the end user API a but uglier

Reply:
For my Swift understanding, could you ellaborate a bit on the overloads causing issues more than default param values @weissi ?

// I'm used to the opposite, where default params are hell since impossible to evolve keeping binary compatibility, yet overloads are safer. A mix of overloads and default params results in hell though (at least in Scala, and reasoning wise I imagine the same way here..), what's a good guideline here -- prefer default params?